feat: add support for ALB

ihendriks opened the pull request:
## Description

AWS ALB support lambdas, serverless supports them as well. As such, supporting them in serverless-offline seems logical.

Based on:
  - https://github.com/dherault/serverless-offline/pull/1250 by @bayoumymac
  - https://github.com/tmaslen/serverless-local-alb-example by @tmaslen

## Motivation and Context

Support for this has been requested by several people.

This would close #598 (which is already closed, but the feature is not supported).

## How Has This Been Tested?

It's tested on our own project with success.

I was unable to get the testsuite to run at this time, but wanted to open the PR for feedback regardless.

ihendriks commented:
@dherault I've rebased the PR and resolved the conflict.

Let me know if this needs further changes from my end. I've been using it locally for our usecase and haven't run into any issues so far.

dnalborczyk commented:
@ihendriks thank you for fixing up the PRs. at a quick glance, it looks great. would be great if we could add some tests.even if you have some examples, I could guide you on how to add those.

ihendriks commented:
@dnalborczyk I've added an example and I've created some unittests. They're mostly duplicated from `integrations/handlers` as that made the most sense to me.

I didn't copy over everything, but figured this was a good starting point.

Let me know if there is anything that needs to be done.

juanitorello commented on src/events/alb/HttpServer.js:
```diff
@@ -0,0 +1,287 @@
+      if (err) {
+        statusCode = errorStatusCode
+      }
+      response.statusCode = statusCode
```
In this case shouldn't be better to have something like this?

```suggestion
      response.statusCode = result.statusCode
```

and it's also missing the headers from the response, so it could be something like this

```suggestion
      response.statusCode = result.statusCode
      response.headers = result.headers
```

dnalborczyk commented:
thank you very much @ihendriks !  I'll try to make some time to have a deeper look very soon. stay tuned. I just wanna iron out some outstanding kinks in v9.

ihendriks commented:
Hi @dnalborczyk, I was wondering what the status of your refactoring is?

If I were to get this PR up to date with the current state of the repo, do you think this could be merged?

I'm reluctant to work on this while you're refactoring at the same time, but I would like to get this merged sooner rather than later, if that's possible.

Not to pressure you or anything, just curious about what the plan/timeline is here.

dnalborczyk commented:
thank you for your patience @ihendriks . 

yes, if you could fix the conflicts that would be great. In the meantime I will wait to change any files related to your PR. If needed, we can flag the **ALB** event support as `experimental` if there's anything to be flashed out further.

ihendriks commented:
Alright, I will try to make some time for it this week.

've fixed some issues that we found ourselves, but I can imagine there's a couple of things I've missed still. So Experimental flag might be a good idea until there's a bit more feedback from other users.

ihendriks commented:
@dnalborczyk Sorry for the delay, was hard to find time for this.

I've resolved all conflicts, fixed a few issues and updated the code to match your refactoring a bit better.

Let me know if there is anything else that needs to be done.

dnalborczyk commented:
thank you for fixing up the PR @ihendriks !!

we should get it merged in the next coming days. I'm thinking to release this as semver major, in case someone had an own/or third party offline plugin for **application load balancer** support, which in turn would likely break things or cause potentially weird behavior - although one could make that argument for _any_ new features.
